core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -321,4 +327,31 @@ public boolean undelete(boolean doUndelete) {
       return false;
    }
 
+   public static class Externalizer extends AbstractExternalizer<ReadCommittedEntry> {
```

Comment:
Function implementations could return the entry view that has been passed in. If that happens in a cluster, that means that the entry view needs to be serialized, and when doing remote operations, seems like the entry view gets built with a read committed entry. I'll check further to see why it gets created with that and not one of the internal cache entry implementations.